src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -207,12 +210,14 @@ def forward(
         key = self._split_heads(key, self.num_attention_heads, self.head_dim, True)
         value = self._split_heads(value, self.num_attention_heads, self.head_dim, False)
 
-        seq_len = key.shape[1]
-        offset = 0
+        if is_torch_fx_proxy(position_ids):
+            embed_positions = get_embed_positions(self.embed_positions, position_ids)
+        else:
+            embed_positions = self._get_embed_positions(position_ids)
```

Comment:
Couldn't the first branch (`embed_positions = get_embed_positions(self.embed_positions, position_ids)`) work in both cases?

It would result in simpler code :)

(ditto for codegen)

Reply:
Thanks @gante, my concern is that the cached position embeddings will initially be in CPU mem and so will need to be copied to GPU every time. The logic in `_get_embed_positions` copies the first time and then no longer needs to, but I couldn't get it to work for the `torch.fx` case, hence this if/else.

I'll add a comment to the code to make that clear. Codegen is slightly simpler because it doesn't support `torch.fx` anyhow.